tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/regex/RegexLineDeserializer.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.tajo.storage.Tuple;
+import org.apache.tajo.storage.text.TextFieldSerializerDeserializer;
+import org.apache.tajo.storage.text.TextLineDeserializer;
+import org.apache.tajo.storage.text.TextLineParsingError;
+import org.apache.tajo.storage.text.TextLineSerDe;
+
+import java.io.IOException;
+import java.nio.charset.CharsetDecoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RegexLineDeserializer extends TextLineDeserializer {
+  private static final Log LOG = LogFactory.getLog(RegexLineDeserializer.class);
+
+  private final CharsetDecoder decoder = CharsetUtil.getDecoder(CharsetUtil.UTF_8);
+  private FieldSerializerDeserializer fieldSerDer;
+  private ByteBuf nullChars;
+
+  private int[] targetColumnIndexes;
+  private String inputRegex;
+  private Pattern inputPattern;
+  // Number of rows not matching the regex
+  private long unmatchedRows = 0;
+  private long nextUnmatchedRows = 1;
+  // Number of rows that match the regex but have missing groups.
+  private long partialMatchedRows = 0;
+  private long nextPartialMatchedRows = 1;
+
+  public RegexLineDeserializer(Schema schema, TableMeta meta, Column[] projected) {
+    super(schema, meta);
+    targetColumnIndexes = PlannerUtil.getTargetIds(schema, projected);
+  }
+
+  @Override
+  public void init() {
+    fieldSerDer = new TextFieldSerializerDeserializer(meta);
+    fieldSerDer.init(schema);
+
+    // Read the configuration parameters
+    inputRegex = meta.getProperty(StorageConstants.TEXT_REGEX);
+    boolean inputRegexIgnoreCase = "true".equalsIgnoreCase(
+        meta.getProperty(StorageConstants.TEXT_REGEX_CASE_INSENSITIVE, "false"));
+
+    // Parse the configuration parameters
+    if (inputRegex != null) {
+      inputPattern = Pattern.compile(inputRegex, Pattern.DOTALL
+          + (inputRegexIgnoreCase ? Pattern.CASE_INSENSITIVE : 0));
+    } else {
+      throw new TajoRuntimeException(new InvalidTablePropertyException(StorageConstants.TEXT_REGEX,
+          "This table does not have serde property \"" + StorageConstants.TEXT_REGEX + "\"!"));
+    }
+
+    if (nullChars != null) {
+      nullChars.release();
+    }
+    nullChars = TextLineSerDe.getNullChars(meta);
+  }
+
+
+  @Override
+  public void deserialize(final ByteBuf lineBuf, Tuple output) throws IOException, TextLineParsingError {
+
+    if (lineBuf == null || targetColumnIndexes.length == 0) {
+      return;
+    }
+
+    String line = decoder.decode(lineBuf.nioBuffer(lineBuf.readerIndex(), lineBuf.readableBytes())).toString();
+    int[] projection = targetColumnIndexes;
+
+    // Projection
+    int currentTarget = 0;
+    int currentIndex = 0;
+    Matcher m = inputPattern.matcher(line);
+
+    if (!m.matches()) {
+      unmatchedRows++;
+      if (unmatchedRows >= nextUnmatchedRows) {
+        nextUnmatchedRows *= 100;
+        // Report the row
+        LOG.warn("" + unmatchedRows + " unmatched rows are found: " + line);
```

Comment:
This also looks good, and I'm just curious why this log is printed repeatedly when 100 unmatched rows are found instead of just printing the total number of unmatched rows when the deserializer is released. Is it more helpful?